core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1002,6 +1001,7 @@ class DAGScheduler(
     val stageId = task.stageId
     val taskType = Utils.getFormattedClassName(task)
 
+    // REVIEWERS: does this need special handling for multiple completions of the same task?
```

Comment:
Doesn't look like it to me.  The OutputCommitCoordinator should deny any completion from other than the currently authorized Stage, and should only authorize one Task per partition to commit per Stage.